src/k4a_ros_device.cpp

```diff
@@ -874,12 +874,16 @@ void K4AROSDevice::framePublisherThread()
   calibration_data_.getRgbCameraInfo(depth_rect_camera_info);
   calibration_data_.getDepthCameraInfo(ir_raw_camera_info);
 
+  const std::chrono::milliseconds firstFrameWaitTime = std::chrono::milliseconds(4 * 1000);
+  const std::chrono::milliseconds regularFrameWaitTime = std::chrono::milliseconds(1000 * 5 / params_.fps);
```

Comment:
This looks like the loop will be 1/5th the rate of params_.fps, any reason for that?

Reply:
For regular frames, the loop will wait up to 5 times the expected amount of time to receive one frame. The wait time is limited so frames are not waited on for an infinite amount of time.

Reply:
During normal operation, we see frame drops, especially if color sensor is active and multiple cameras are connected. Waiting UP TO 5 times the time of a frame would need to arrive will not crash the driver on any frame drop.
Also note, this is a timeout, the function will return as soon as a new frame arrives.

Reply:
During normal operation, we see frame drops, especially if color sensor is active and multiple cameras are connected. Waiting UP TO 5 times the time of a frame would need to arrive will not crash the driver on any frame drop.
Also note, this is a timeout, the function will return as soon as a new frame arrives.